### ISSUE TITLE
Can't change firing mode with double-barrel shotgun and birdshots

### ISSUE THREAD

**Issue by Archos7** [cataclysmbnteam/Cataclysm-BN#4071]
### Describe the bug

A double-barrel shotgun have 2 firing modes: single and double. However, when equipped with birdshots, you can't change the fire mode, defaulting first to single shot mode.

Equipping 00 shots allows for firing mode to change and work properly, though. Also, after changing to double with 00's and going back to birdshot, birdshots are shot as double, so i imagine it's intended for them to be shot with both fire types.

### Steps To Reproduce

1. Make a player with double-barrel shotgun, 00 shots and birdshots.
2. Load birdshots on shotgun. Fire mode can't be changed and will shot as single.
3. Load 00s on shotgun
4. Change fire mode to double with the 00s, which is allowed
5. Fire the shotgun
6. Reload shotgun with birdshots and fire
7. Shotgun will shot both birdshots cartridges at once, but still doesn't allow for firing mode to be changed

### Screenshots

![Clipboard03](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/1795259/5cf5ed76-87ce-47c1-a7ba-c874f48ea184)
![Clipboard04](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/1795259/51d82abf-72b2-4353-82c7-0fdb06c73b67)
![Clipboard05](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/1795259/4cef6103-494e-40bb-b6ac-41363a4a1cc0)
![Clipboard01](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/1795259/35883035-4cca-4f21-b69f-8b5b106ed0e3)
![Clipboard02](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/1795259/9ed103c8-5916-40f5-9bae-7dcbbe9d456a)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3803 (22H2)
- Game Version: 0.5 [64-bit]
- Graphics Version: Tiles
- LAPI Version: 1
- Game Language:  []
- Mods loaded: [
    Bright Nights [bn],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    No Rail Stations [No_Rail_Stations],
    Prevent Zombie Revivication [no_reviving_zombies],
    Limit Fungal Growth [limit_fungal_growth]
]

### Additional context

Also happens with double-barrel pipe shotgun.

**Comment by RoyalFox2140** [comment]
I'm using (F) to toggle attack mode of wielded item, and I can switch fire modes on any ammunition. Tested with modded shotgun and with double barrel shotgun, bird shot and 00 shot on both. This is on MSVC build of 0.5 stable.

![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/112293514/6e448cc7-1c1f-4bfd-bf05-2b8eff3a8ce9)
![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/112293514/71e8c644-c793-442e-b22e-415af85e1807)
![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/112293514/64371b83-2ec1-4c2a-86ce-f12452f3c4ca)

- OS: Windows
    - OS Version: 10.0.19045.3448 (22H2)
- Game Version: 0.5 [64-bit]
- Graphics Version: Tiles
- LAPI Version: 1
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [bn],
    Craftable Gun Pack [craftgp],
    Alternative Map Key [alt_map_key],
    Craftable Bullet Casings [CasingCrafting],
    MST Extra [MST_Extra],
    Essence 2200 BN [Essence2200_BN],
    Essence 2200 Tweaks [FoxTweaks],
    Nonperishable Overhaul [Nonperishable_Overhaul],
    No Addictions [No_Addictions],
    Limit Fungal Growth [limit_fungal_growth],
    Disable NPC Needs [no_npc_food],
    Simplified Nutrition [novitamins],
    Mining Mod [Mining_Mod],
    Rural-Only Mapgen [ruralbiome],
    No Monsters [Only_Wildlife]
]

**Comment by Archos7** [comment]
You're right RoyalFox, I didn't knew we could change fire mode by pressing shift + f out of aiming mode.

In aiming mode, however, it still doesn't change the firing mode when using birdshots and pressing s, just tested here. Does not show the option for that on the aiming menu as well. I can do that when using 00s however.

I guess it's something that happens only on aiming mode i guess. 

00s: 
![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/1795259/3a2e18ea-b582-460b-b642-5cda28544a14)

birdshot:
![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/1795259/74b9302b-d52e-414c-bb09-d6acf7901ed4)

**Comment by RoyalFox2140** [comment]
Maybe its because you can't aim birdshot. It's guaranteed to hit because its an AoE cone attack.
